### PR TITLE
Improve printing of modules

### PR DESCRIPTION
**Pull request by pfultz2**
This makes several improvements:

1. Only print target_id when its not 0.
2. Only prefix variable names if there is no names in an outer scope. This improves the printing when using `debug_print` so the names wont have a long module name prepended to them.
3. Add the shape information as a comment when printing python.
4. Use `create_argument` in python for small literals instead of always using `generate_argument`.

### CODE REVIEW

**Comment by causten** [comment]
Can you add a before / after with this change to this PR

**Comment by pfultz2** [comment]
before:
```
x = @param:x -> float_type, {2, 32, 327680}, {10485760, 327680, 1}, target_id=0
@1 = @literal{327680} -> float_type, {1}, {1}, target_id=0
@2 = multibroadcast[out_lens={2, 32, 327680},out_dyn_dims={}](@1) -> float_type, {2, 32, 327680}, {0, 0, 0}, target_id=0
@3 = mul(x,@2) -> float_type, {2, 32, 327680}, {10485760, 327680, 1}, target_id=0
@4 = reduce_sum[axes={2}](@3) -> float_type, {2, 32, 1}, {32, 1, 1}, target_id=0
@5 = mul(x,x) -> float_type, {2, 32, 327680}, {10485760, 327680, 1}, target_id=0
@6 = mul(@5,@2) -> float_type, {2, 32, 327680}, {10485760, 327680, 1}, target_id=0
@7 = reduce_sum[axes={2}](@6) -> float_type, {2, 32, 1}, {32, 1, 1}, target_id=0
@8 = mul(@4,@4) -> float_type, {2, 32, 1}, {32, 1, 1}, target_id=0
@9 = sub(@7,@8) -> float_type, {2, 32, 1}, {32, 1, 1}, target_id=0
@10 = add(@4,@9) -> float_type, {2, 32, 1}, {32, 1, 1}, target_id=0
@11 = @return(@10), target_id=0
```

after:
```
module: "main"
x = @param:x -> float_type, {2, 32, 327680}, {10485760, 327680, 1}
@1 = @literal{327680} -> float_type, {1}, {1}
@2 = multibroadcast[out_lens={2, 32, 327680},out_dyn_dims={}](@1) -> float_type, {2, 32, 327680}, {0, 0, 0}
@3 = mul(x,@2) -> float_type, {2, 32, 327680}, {10485760, 327680, 1}
@4 = reduce_sum[axes={2}](@3) -> float_type, {2, 32, 1}, {32, 1, 1}
@5 = mul(x,x) -> float_type, {2, 32, 327680}, {10485760, 327680, 1}
@6 = mul(@5,@2) -> float_type, {2, 32, 327680}, {10485760, 327680, 1}
@7 = reduce_sum[axes={2}](@6) -> float_type, {2, 32, 1}, {32, 1, 1}
@8 = mul(@4,@4) -> float_type, {2, 32, 1}, {32, 1, 1}
@9 = sub(@7,@8) -> float_type, {2, 32, 1}, {32, 1, 1}
@10 = add(@4,@9) -> float_type, {2, 32, 1}, {32, 1, 1}
@11 = @return(@10)
```

before:
```python
p = migraphx.program()
m = p.get_main_module()
p_x = m.add_parameter("x",migraphx.shape(type="float_type", lens=[2,32,327680]))
x_1 = m.add_literal(migraphx.generate_argument(migraphx.shape(type="float_type", lens=[1]), 0))
x_2 = m.add_instruction(migraphx.op("multibroadcast", out_lens=[2,32,327680]), [x_1])
x_3 = m.add_instruction(migraphx.op("mul"), [p_x, x_2])
x_4 = m.add_instruction(migraphx.op("reduce_sum", axes=[2]), [x_3])
x_5 = m.add_instruction(migraphx.op("mul"), [p_x, p_x])
x_6 = m.add_instruction(migraphx.op("mul"), [x_5, x_2])
x_7 = m.add_instruction(migraphx.op("reduce_sum", axes=[2]), [x_6])
x_8 = m.add_instruction(migraphx.op("mul"), [x_4, x_4])
x_9 = m.add_instruction(migraphx.op("sub"), [x_7, x_8])
x_10 = m.add_instruction(migraphx.op("add"), [x_4, x_9])
m.add_return([x_10])
```

after:
```python
p = migraphx.program()
m = p.get_main_module()
p_x = m.add_parameter("x", migraphx.shape(type="float_type", lens=[2, 32, 327680]))
x_1 = m.add_literal(migraphx.create_argument(migraphx.shape(type="float_type", lens=[1]), [327680]))
x_2 = m.add_instruction(migraphx.op("multibroadcast", out_lens=[2,32,327680]), [x_1]) # migraphx.shape(type="float_type", lens=[2, 32, 327680], strides=[0, 0, 0])
x_3 = m.add_instruction(migraphx.op("mul"), [p_x, x_2]) # migraphx.shape(type="float_type", lens=[2, 32, 327680])
x_4 = m.add_instruction(migraphx.op("reduce_sum", axes=[2]), [x_3]) # migraphx.shape(type="float_type", lens=[2, 32, 1])
x_5 = m.add_instruction(migraphx.op("mul"), [p_x, p_x]) # migraphx.shape(type="float_type", lens=[2, 32, 327680])
x_6 = m.add_instruction(migraphx.op("mul"), [x_5, x_2]) # migraphx.shape(type="float_type", lens=[2, 32, 327680])
x_7 = m.add_instruction(migraphx.op("reduce_sum", axes=[2]), [x_6]) # migraphx.shape(type="float_type", lens=[2, 32, 1])
x_8 = m.add_instruction(migraphx.op("mul"), [x_4, x_4]) # migraphx.shape(type="float_type", lens=[2, 32, 1])
x_9 = m.add_instruction(migraphx.op("sub"), [x_7, x_8]) # migraphx.shape(type="float_type", lens=[2, 32, 1])
x_10 = m.add_instruction(migraphx.op("add"), [x_4, x_9]) # migraphx.shape(type="float_type", lens=[2, 32, 1])
m.add_return([x_10])
```

**Review by umangyadav — APPROVED** [review]

**Comment by umangyadav** [comment]
@causten  do not merge PR if any of the CI checks are failing.  Let us investigate failures before merging. 
Licensing was failing on this PR.  I had to put a patch for that on my PR here https://github.com/ROCm/AMDMIGraphX/pull/3082

**Pull request merged**
